### PR TITLE
adding trigger for steammode

### PR DESCRIPTION
**Pull request by LoQue90**
Steam switch has now 3 modes:
off, switch or trigger
removed double pinmode declaration for Steamswitch
renaming powerswitchvoid

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -2239,11 +2233,13 @@ void looppid() {
     checkSteamON();
     setEmergencyStopTemp();
     checkpowerswitch();
-    handleMachineState();
+    checksteamswitch();
```
Comment (kjyv) [inline src/main.cpp:2236]:
```suggestion
    checkSteamSwitch();
```

File: src/main.cpp [hunk 2]
```diff
@@ -2239,11 +2233,13 @@ void looppid() {
...
 
     if (standbyModeOn && machineState != kStandby) {
         updateStandbyTimer();
     }
+    
+    handleMachineState();      // update machineState
```
Comment (kjyv) [inline src/main.cpp:2242]:
```suggestion
    handleMachineState(); 
```

File: src/steamswitchvoid.h [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                        steamON = 1;
+                    } else {
+                        Serial.println("Turn Steam OFF");
+                        steamON = 0;
+                    }
+                }
+            }
+        }
+
+        laststeamswitchTrigger = reading;
```
Comment (kjyv) [inline src/steamswitchvoid.h:58]:
```suggestion
        lastSteamSwitchTrigger = reading;
```

File: src/steamswitchvoid.h [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+         // if activated via web interface then steamFirstON == 1, prevent override
+        if (digitalRead(PIN_STEAMSWITCH) == LOW && steamFirstON == 0) {
+            steamON = 0;
+        }
+    #endif
+
+    #if STEAMSWITCHTYPE == 2  // TRIGGER
+        int reading = digitalRead(PIN_STEAMSWITCH);
+
+        if (reading != laststeamswitchTrigger) {
```
Comment (kjyv) [inline src/steamswitchvoid.h:32]:
```suggestion
        if (reading != lastSteamSwitchTrigger) {
```

File: src/steamswitchvoid.h [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+ * @file steamswitchvoid.h
+ *
+ * @brief
+ */
+
+/**
+ * @brief Digtalswitch input pin for STEAM SWITCH
+ */
+
+int laststeamswitchTrigger = LOW;                   // the previous reading from the input pin
```
Comment (kjyv) [inline src/steamswitchvoid.h:11]:
```suggestion
int lastSteamSwitchTrigger = LOW;                   // the previous reading from the input pin
```

**Review by kjyv — APPROVED** [review]
some renames as suggested, otherwise approved from my side

**Review by fiendie — APPROVED** [review]

**Review by kjyv — APPROVED** [review]

**Pull request merged**